### DISCUSSION TITLE
With the SDK level 29 the NDK 25 binder_auto_utils.h raises error

### DISCUSSION THREAD

**Discussion by MMeram** [android/ndk#1881] [General]
In our AAR configuration we use a native library and a set of AIDL interfaces. I noticed probably one error in the 'binder_auto_utils.h'
`std::string getDescription() const` method that manifests itself as a compile error if you compile with the minimum SDK version 29.
Can you check if it should look like it like this:

```c++
std::string getDescription() const {
  #if __ANDROID_API__ >= 30
  #ifdef __ANDROID_UNAVAILABLE_SYMBOLS_ARE_WEAK__
          if (__builtin_available(android 30, *)) {
  #else
          if (__ANDROID_API__ >= 30) {
  #endif
              const char* cStr = AStatus_getDescription(get());
              std::string ret = cStr;
              AStatus_deleteDescription(cStr);
              return ret;
          }
  #endif
 (...)
 }
```

In the original code this code snippet always fails to compile if you use minSDKVersion < 30 like 29 since AStatus_getDescription() and AStatus_deleteDescription() only available with the API level >= 30

**Comment by DanAlbert** [comment]
You may need to use the binder_auto_utils.h that's distributed in build-tools. This header was only added to the NDK by mistake. It may have the same problem, but the fix as far as the NDK is concerned is to delete the header, because it's actually an aapt header, not an NDK header.

Also, use:

    ```c++
    code
    ```

To make a code snippet (replace "c++" with whatever language you're pasting). `>` makes a block quote and makes an unreadable code snippet. Use \``inline code`\` for inline snippets.

**Comment by MMeram** [comment]
Unfortunately, this header is not available in build-tools and the NDK 25 enforces the build process in a way that building with the API level 29 is not possible. In the short run I can patch this header like indicated above. Would you consider it as a bug or do you have any other solution proposal?
Thank you

**Reply by DanAlbert** [reply]
It is supposed to be. https://issuetracker.google.com/219987524#comment3 (you'll have better luck asking your question in that bug component than here, we don't have the answers)

**Comment by MMeram** [comment]
This link doesn't work for me. Anyways, we can get along with this workaround. I wonder though if anyone else had the same issue.

**Reply by DanAlbert** [reply]
Fixed. You know you can reply to the thread, right? You're creating a new thread with each comment. You want the "write a reply" box, not "Write a comment".
